core/src/Kokkos_CopyViews.hpp

```diff
@@ -2117,6 +2117,26 @@ create_mirror_view_and_copy(const Space& , const Kokkos::View<T,P...> & src
   deep_copy(mirror, src);
   return mirror;
 }
+    
+// Create a mirror view in a new space without initializing (specialization for same space)
+template<class Space, class T, class ... P>
+typename Impl::MirrorViewType<Space,T,P ...>::view_type
+create_mirror_view(const Space& , const ViewAllocateWithoutInitializing & arg_prop
+  , const Kokkos::View<T,P...> & src
+  , typename std::enable_if<Impl::MirrorViewType<Space,T,P ...>::is_same_memspace>::type* = 0 ) {
+  return src;
+}
+    
+// Create a mirror view in a new space without initializing (specialization for different space)
+template<class Space, class T, class ... P>
+typename Impl::MirrorViewType<Space,T,P ...>::view_type
+create_mirror_view(const Space& , const ViewAllocateWithoutInitializing & arg_prop
```

Comment:
I think this is getting at the right idea.  I would be tempted to put the tag last, like this:
```c++
template<class Space, class DataType, class ... P>
typename Impl::MirrorViewType<Space, DataType, P...>::view_type
create_mirror_view (const Space& memSpace, 
  const Kokkos::View<DataType, P...>& src, 
  Kokkos::Impl::WithoutInitializing_t);
```
so that users can call it like this:
```c++
auto x_mirror = create_mirror_view (SomeMemorySpace (), x, WithoutInitializing);
```
I realized that it doesn't make sense to pass in the whole ViewCtorProp thing, because not all the properties make sense.  (For example, you wouldn't want `create_mirror_view` to create a View with different padding than its input.)  This overloading approach would allow for future expansion, if perhaps other properties make sense.